src/main/target/PIKOBLX_limited/target.c

```diff
@@ -23,12 +23,12 @@
 #include "drivers/timer.h"
 
 const timerHardware_t timerHardware[USABLE_TIMER_CHANNEL_COUNT] = {
-    { TIM3,  IO_TAG(PA4),  TIM_Channel_2, 1, IOCFG_AF_PP,    GPIO_AF_2, TIM_USE_MC_MOTOR }, // PWM1  - PA4  - *TIM3_CH2
-    { TIM3,  IO_TAG(PA6),  TIM_Channel_1, 1, IOCFG_AF_PP,    GPIO_AF_2, TIM_USE_MC_MOTOR }, // PWM2  - PA6  - *TIM3_CH1, TIM8_BKIN, TIM1_BKIN, TIM16_CH1
-    { TIM3,  IO_TAG(PB0),  TIM_Channel_3, 1, IOCFG_AF_PP,    GPIO_AF_2, TIM_USE_MC_MOTOR }, // PWM3  - PB0  - *TIM3_CH3, TIM1_CH2N, TIM8_CH2N
-    { TIM3,  IO_TAG(PB1),  TIM_Channel_4, 1, IOCFG_AF_PP,    GPIO_AF_2, TIM_USE_MC_MOTOR }, // PWM4  - PB1  - *TIM3_CH4, TIM1_CH3N, TIM8_CH3N
-    { TIM2,  IO_TAG(PA1),  TIM_Channel_2, 1, IOCFG_AF_PP,    GPIO_AF_1, TIM_USE_MC_MOTOR }, // PWM5  - PA1  - *TIM2_CH2, TIM15_CH1N
-    { TIM2,  IO_TAG(PA2),  TIM_Channel_3, 1, IOCFG_AF_PP,    GPIO_AF_1, TIM_USE_MC_MOTOR }, // PWM6  - PA2  - *TIM2_CH3, !TIM15_CH1
+    { TIM3,  IO_TAG(PA4),  TIM_Channel_2, 1, IOCFG_AF_PP,    GPIO_AF_2, TIM_USE_MC_MOTOR | TIM_USE_FW_MOTOR }, // PWM1  - PA4  - *TIM3_CH2
+    { TIM3,  IO_TAG(PA6),  TIM_Channel_1, 1, IOCFG_AF_PP,    GPIO_AF_2, TIM_USE_MC_MOTOR | TIM_USE_FW_MOTOR }, // PWM2  - PA6  - *TIM3_CH1, TIM8_BKIN, TIM1_BKIN, TIM16_CH1
+    { TIM3,  IO_TAG(PB0),  TIM_Channel_3, 1, IOCFG_AF_PP,    GPIO_AF_2, TIM_USE_MC_MOTOR | TIM_USE_FW_SERVO }, // PWM3  - PB0  - *TIM3_CH3, TIM1_CH2N, TIM8_CH2N
```

Comment:
This combination might cause unexpected behavior. In general same timer (in our case TIM3) can only be shared between servo and motor only if PWM frequency is the same - in our case 50Hz. We need to enforce that user doesn't select PWM-400Hz or OneShot125 or any other protocol. Same for servos - only Servos at 50Hz would be reliably supported.

Reply:
I understand. With this config even if the ESC frequency is set to something else than 50Hz the output is 50Hz. I looked at the doc and unfortunately is looks like we can't have a different timer for PWM1-2 and PWM3-6 because of the pins chosen as PWM outputs. If I understand correctly the datasheet PWM1(PA4)/PWM2(PA6)/PWM3(PB0)/PWM4(PB1) can only use timer 3 so we can't use a different timer for PWM1/2 and PWM3/4. The only solution would be to use the current PWM5(PA1)/PWM6(PA2) pins as PWM1/2 because they can use timer 2. Can you confirm ?

I don't see any reason why it couldn't be done (swapping PWM1/2 with PWM5/6 pins) if this target isn't used in pre-wired crafts. The documentation would just need to be explicit about the remapping. The silkscreen on the boards I got for PWM5/6 is already wrong anyway.